### PR TITLE
[DX-1158, TT-11589] Added API category docs

### PR DESCRIPTION
**Pull request by andyo-tyk**
### Preview Link
https://deploy-preview-4215--tyk-docs.netlify.app/docs/nightly/product-stack/tyk-dashboard/advanced-configurations/api-categories/

### Description
Added documentation of the API categories feature

___

## **Type**
documentation, enhancement

___

## **Description**
- Introduced comprehensive documentation for the API Categories feature in Tyk Dashboard, covering its purpose, use cases, and management instructions.
- Added a new section in the Tyk Dashboard documentation for API Categories, explaining how to use this feature for both Tyk OAS and Tyk Classic APIs.
- Updated the navigation menu to include a link to the new API Categories documentation, making it easily accessible to users.

___

## **Changes walkthrough**
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Documentation</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>api-categories.md</strong><dd><code>New Documentation for API Categories Feature</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

tyk-docs/content/product-stack/tyk-dashboard/advanced-configurations/api-categories.md
<li>Introduced a new documentation page for API Categories in Tyk <br>Dashboard.<br> <li> Explained the concept, use cases, and how to manage API categories.<br> <li> Provided guidance for both Tyk OAS and Tyk Classic APIs regarding <br>category management.<br> <li> Included notes on behavior during API migration with Tyk Sync.

</details>
    

  </td>
  <td><a href="https://github.com/TykTechnologies/tyk-docs/pull/4215/files#diff-f07e7ef6da9026d309b98cc5a35b7e2c1de4ae7e21977deb24f32e847819544b">+101/-0</a>&nbsp; </td>
</tr>                    
</table></td></tr><tr><td><strong>Configuration changes</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>menu.yaml</strong><dd><code>Updated Navigation Menu to Include API Categories</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

tyk-docs/data/menu.yaml
<li>Added a new menu entry for the API Categories documentation under the <br>Tyk Dashboard section.

</details>
    

  </td>
  <td><a href="https://github.com/TykTechnologies/tyk-docs/pull/4215/files#diff-a6789c66bfa0660f7fdbe4796002ab4777677bae59a0276689c840100ff0b0b2">+4/-0</a>&nbsp; &nbsp; &nbsp; </td>
</tr>                    
</table></td></tr></tr></tbody></table>

___

> ✨ **PR-Agent usage**:
>Comment `/help` on the PR to get a list of all available PR-Agent tools and their descriptions

### CODE REVIEW

File: tyk-docs/content/product-stack/tyk-dashboard/advanced-configurations/api-categories.md [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+{{< img src="/img/dashboard/endpoint-designer/categories-filter.png" alt="View APIs in a category" >}} 
+
+### Tyk Dashboard API
+The [Tyk Dashboard API]({{< ref "tyk-dashboard-api" >}}) provides endpoints to manage categories directly, if you are not using the API Designer.
+
+When working with Tyk OAS APIs, you can manage categories for an API using these endpoints:
+
+| Method | Endpoint path                        | Action                                                                                 |
+|--------|--------------------------------------|----------------------------------------------------------------------------------------|
+| `PUT`  | `/api/apis/oas/{apiId}/categories`   | Assign a list of categories to the specified API                                       |
```
Comment (dcs3spp) [inline tyk-docs/content/product-stack/tyk-dashboard/advanced-configurations/api-categories.md]:
```suggestion
| `PUT`  | `/api/apis/oas/{apiID}/categories`   | Assign a list of categories to the specified API   
```

File: tyk-docs/content/product-stack/tyk-dashboard/advanced-configurations/api-categories.md [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+| `GET`  | `/api/apis/oas/{apiID}/categories`   | Retrieve the list of categories assigned to the specified API                          |
+
+When working with Tyk Classic APIs, you manage categories for an API by modifying the `name` field in the API definition and then updating the API in Tyk with that using these endpoints:
+
+| Method | Endpoint                             | Action                                                                                 |
+|--------|--------------------------------------|----------------------------------------------------------------------------------------|
+| `PUT`  | `/api/apis/{apiId}`                  | Update the API definition for the specified API - CRUD category tags in the `name` field |
```
Comment (dcs3spp) [inline tyk-docs/content/product-stack/tyk-dashboard/advanced-configurations/api-categories.md]:
```suggestion
| `PUT`  | `/api/apis/{apiID}`                  | Update the API definition for the specified API - CRUD category tags in the `name` field |
```

File: tyk-docs/content/product-stack/tyk-dashboard/advanced-configurations/api-categories.md [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+The [Tyk Dashboard API]({{< ref "tyk-dashboard-api" >}}) provides endpoints to manage categories directly, if you are not using the API Designer.
+
+When working with Tyk OAS APIs, you can manage categories for an API using these endpoints:
+
+| Method | Endpoint path                        | Action                                                                                 |
+|--------|--------------------------------------|----------------------------------------------------------------------------------------|
+| `PUT`  | `/api/apis/oas/{apiId}/categories`   | Assign a list of categories to the specified API                                       |
+| `GET`  | `/api/apis/oas/{apiID}/categories`   | Retrieve the list of categories assigned to the specified API                          |
+
+When working with Tyk Classic APIs, you manage categories for an API by modifying the `name` field in the API definition and then updating the API in Tyk with that using these endpoints:
```
Comment (dcs3spp) [inline tyk-docs/content/product-stack/tyk-dashboard/advanced-configurations/api-categories.md:88]:
```suggestion
When working with Tyk Classic APIs, you can manage categories for an API using these endpoints:
```

File: tyk-docs/content/product-stack/tyk-dashboard/advanced-configurations/api-categories.md [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+
+## When to use API categories
+#### Managing a large portfolio of APIs
+As a platform manager looking after a large portfolio of APIs, if I need to make changes to a sub-set of APIs, it's cumbersome having to identify which APIs they are and then to find them one-by-one in the list. If I have assigned categories to my APIs then I can filter quickly and easily to work with that sub-set. What's really powerful is that an API can appear in as many different categories as I like. 
+
+#### Multi-tenant deployment
+If you are a super-user on a multi-tenant deployment - the admin user where [role-based access control]({{< ref "tyk-dashboard/rbac" >}}) is being used to give different users or teams of users access to a sub-set of the entire API portfolio - then by aligning categories with the API ownership rules that you have deployed you will be able to filter the list of APIs to those visible to each separate user group/team.
```
Comment (dcs3spp) [inline tyk-docs/content/product-stack/tyk-dashboard/advanced-configurations/api-categories.md]:
```suggestion
Multi-tenant deployments with [role-based access control]({{< ref "tyk-dashboard/rbac" >}}) enabled allows an admin user to give different users or groups access to a sub-set of the entire API portfolio. Categories can be aligned with the API ownership rules that you have deployed to allow filtering the list of APIs for those visible to each separate user group/team.
```

**Review by dcs3spp — CHANGES_REQUESTED** [review]
@andyo-tyk PR LGTM, I've added minor suggestions. @andrei-tyk would you be able to add a peer review for Andy?

File: tyk-docs/content/product-stack/tyk-dashboard/advanced-configurations/api-categories.md (hunk 3 above)
Comment (andyo-tyk) [reply tyk-docs/content/product-stack/tyk-dashboard/advanced-configurations/api-categories.md:88]: This information is important so I'd rather leave as-is.

**Review by andrei-tyk — APPROVED** [review]

**Review by dcs3spp — APPROVED** [review]

**Pull request merged**